Extended instructions on provisioning

matejart opened the pull request:
We've expanded the instructions to include the description and use of the PXE Server functionality of the `{product-title}`.

matejart committed:
Add description of PXE server management

PXE servers are essential for bare metal provisioning, therefore
we provide our own take on the management of the PXE Servers in
the ManageIQ/CloudForms.

miha-plesko commented on redfish_guide/topics/prerequisites.adoc:
```diff
@@ -10,7 +10,8 @@ The following preprequisites must be met to start using {redfish-provider-full}:
   (SSE) technology.
 * One or more functional PXE servers must be configured and running on a network
-  that is accessible both to {product-appliance}. The PXE servers need to be
+  that is accessible both to {product-appliance}. They need to offer SAMBA or
```
you say "both to" but then you only list one

miha-plesko commented on redfish_guide/topics/using-redfish-physical-infra.adoc:
```diff
@@ -135,6 +135,92 @@ PXE server with the network boot parameters specified in the request for the
+  require a TFTP access for reading the initial executable file containing the
+  PXE itself, and the HTTP access for downloading the Operating Systems' files.
+
+The process of configuring the PXE Servers is similar to that explained in the
```
Not sure why go with plural "PXE servers". I'd stick to singular

miha-plesko commented on redfish_guide/topics/using-redfish-physical-infra.adoc:
```diff
@@ -135,6 +135,92 @@ PXE server with the network boot parameters specified in the request for the
+  PXE itself, and the HTTP access for downloading the Operating Systems' files.
+
+The process of configuring the PXE Servers is similar to that explained in the
+*PXE Provisioning* section of the *Provisioning Virtual Machines and Instances*
```
I'd go with 

```
<<PXE Provisioning>> section of the <<Provisioning Virtual Machines and Instances>>
```

so that PDF will crate links. In our local document the links won't work because we don't have those sections, but when they put it all together it _should_ just work

miha-plesko replied:
Or not, as @tadeboro suggested there is no "one big PDF" available but rather number of small PDFs and you're refering to https://access.redhat.com/documentation/en-us/red_hat_cloudforms/5.0-beta/pdf/provisioning_virtual_machines_and_instances/Red_Hat_CloudForms-5.0-Beta-Provisioning_Virtual_Machines_and_Instances-en-US.pdf so the reference can not work really.

miha-plesko commented on redfish_guide/topics/using-redfish-physical-infra.adoc:
```diff
@@ -135,6 +135,92 @@ PXE server with the network boot parameters specified in the request for the
+We visit *Compute -> Infrastructure -> PXE*. Then, we click *Configuration ->
+Add a New PXE server*. We fill in the fields to describe the server.
+
+image::docs_redfish_using22_new_pxe_server_dialog.png[alt="New PXE Server dialog",512,align="center"]
```
![image](https://user-images.githubusercontent.com/8102426/62940531-171cf180-bdd4-11e9-91fc-7037afdcc642.png)

I'm not sure PXE dir and Customization dir are supposed to be the same because in theory you could accidently name your customization template as `default` which would actually override your PXE menu definition.

tadeboro replied:
They are the same on our top-of-the-rack appliance setup. And the danger of overwriting an existing file is almost non-existent since customization templates that are put on the PXE server have names like `12:34:56:78:90:ab.cfg.ks`.

But I agree that maybe we should show give those two folders different names here since content of the first directory is expected to accessible over NFS/SAMBA and the customization directory over access url.

miha-plesko reviewed (commented):
I'm not too familiar with PXE but I think we shouldn't instruct user to use same directory for customization and PXE menu directory. I understand customization directory is where we write customization template per MAC address, but what is PXE direcotry anyway? I mean, the PXE menu file technically doesn't really need to be in any directory, right @tadeboro ?

tadeboro commented:
Just a quick comment about the PXE server data. When we are registering our PXE server with MIQ, we have no choice as to what goes in each filed. Our PXE server is configured in a certain way and all we can do is input the right data into the dialog. If the PXE server is configured in such a way that PXE and customization directory are the same, then so be it.

tadeboro commented on redfish_guide/topics/prerequisites.adoc:
```diff
@@ -10,7 +10,8 @@ The following preprequisites must be met to start using {redfish-provider-full}:
 * One or more functional PXE servers must be configured and running on a network
-  that is accessible both to {product-appliance}. The PXE servers need to be
+  that is accessible both to {product-appliance}. They need to offer SAMBA or
+  NFS write access, and read-only TFTP and HTTPS access. They also need to be
```
HTTPS is just an example here since we can use just about any protocol we wish here. The only requirement is that the installation program is capable of accessing the customization template over this protocol. Maybe we should avoid talking about protocols here and simply state that MIQ needs to be able to access to the PXE server and put the access details into later section?

tadeboro commented on redfish_guide/topics/using-redfish-physical-infra.adoc:
```diff
@@ -135,6 +135,92 @@ PXE server with the network boot parameters specified in the request for the
 servers defined in the request. It then restarts these servers into one-time
 network boot, initiating the provisioning.
 
+==== Configuring PXE
```
We are not really configuring PXE. We are adding/registering an existing PXE server to the ManageIQ.

tadeboro commented on redfish_guide/topics/using-redfish-physical-infra.adoc:
```diff
@@ -135,6 +135,92 @@ PXE server with the network boot parameters specified in the request for the
 
+==== Configuring PXE
+
+In {product-title}, a PXE server represents a collection of services and files
```
PXE server in MIQ is actually just a bag of connection details. All of the services that enable us to boot over a network are part of the actual PXE server and ManageIQ has nothing to do with them. As far as MIQ is concerned, a PXE server is just an RW depot. ManageIQ does not utilize anything else.

tadeboro reviewed (changes requested):
I feel uneasy about the current state of this document because it feels like there is no way one could register an existing PXE server to MIQ and expect provisioning process to work using only instructions in this document.

I think we need to substantially expand the prerequisites part and explicitly state the assumptions about the PXE server that we will make when describing the registration process (what needs to be accessible from MIQ, what is only accessed by the server we are provisioning, protocols involved, ...). We do not need to give instruction on *how* to set things up, just clearly state *what** we expect to hold.

I will try to write something down later today to get the things rolling.

Commit:
Improve the PXE descriptions

Better explain the prerequisites and the parameters used when
registering a PXE Server in {product-appliance}.

matejart commented:
Thanks for comments and contributions!
